src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PENamedTypeSymbol.cs

```diff
@@ -366,8 +377,174 @@ public override ExtendedSpecialType ExtendedSpecialType
             }
         }
 
-        // PROTOTYPE metadata is undone
-        internal sealed override ParameterSymbol ExtensionParameter => throw new NotImplementedException();
+        internal sealed override ParameterSymbol ExtensionParameter
+        {
+            get
+            {
+                if (!IsExtension)
+                {
+                    return null;
+                }
+
+                var uncommon = GetUncommonProperties().lazyExtensionInfo;
+
+                if (uncommon.LazyExtensionParameter is null)
+                {
+                    var extensionParameter = makeExtensionParameter();
+                    Interlocked.CompareExchange(ref uncommon.LazyExtensionParameter, new StrongBox<ParameterSymbol>(extensionParameter), null);
+                }
+
+                return uncommon.LazyExtensionParameter.Value;
+
+                ParameterSymbol makeExtensionParameter()
```

Comment:
nit: consider making `static` to avoid captures, or limiting capture to `this`. Same for `getMarkerMethodSymbol` below